Signup Frontend

Lucheti29 opened the pull request:
## Summary

Signup layout with input checks added (the layout is not done yet)
## Test
- Let the fields empty and click on Sign up. 
- Put an email with invalid format and click on Sign up. 
- Put different passwords and click on Sign up.
## Screenshot

![Signup Layout](http://s1.postimg.org/7nwi6mpwf/signup.png)
## Card

https://trello.com/c/U8tZwPwz/15-signup

Lucheti29 committed:
-Login layout added
-colors.xml added
-Login strings added
-Login assets added
-Implementation to check empty and valid fields added
-Handler to start Login or Board activity (depends if the user is logged) implemented

Lucheti29 committed:
-Link to Terms and conditions added
-Login image on top added

Lucheti29 committed:
-Signup layout with fields check added
-Signup assets added

Note: the layout is not done yet

Lucheti29 committed:
-String correction

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Starter.java:
```diff
@@ -1,9 +1,10 @@
+import ar.com.wolox.lucasdelatorre.training.activities.Login;
 
 
 public class Starter extends ActionBarActivity {
```
No solemos usar una activity starter. Casi siempre usamos la de login como inicial. Para que lo tengas en cuenta ahora dejalo con esta no hay drama.

Lucheti29 replied:
Es la única forma que se me ocurrió en el momento de hacerlo, si bien no me gustaba cómo quedó. A medida que vaya avanzando lo toco para que quede bien.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Starter.java:
```diff
@@ -11,29 +12,21 @@
         setContentView(R.layout.activity_starter);
-    }
 
+        if(!isLogged())
```
Para que esta este if si siempre va a devolver false?. En un futuro vas a cambiar el metodo isLogged()?

jpdelorenzo replied:
Por otro lado los if los hacemos con este estilo. 

if (sentencia) { 
.....
}

Lucheti29 replied:
1) Sí, era para dejar el método para una futura implementación. Es false así inicia la Activity de Login.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Starter.java:
```diff
@@ -11,29 +12,21 @@
-        // Inflate the menu; this adds items to the action bar if it is present.
-        getMenuInflater().inflate(R.menu.menu_starter, menu);
-        return true;
+        finish();
```
Esto estaria de más. Nunca va a llegar acá.

Lucheti29 replied:
Si mal no recuerdo, el finish() era para el Starter. Cuando cambie eso, se va el método.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Utils.java:
```diff
@@ -0,0 +1,34 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
```
Sacate esta conf del android studio que te agrega estos comments.

Lucheti29 replied:
¿Te referís a estos comentarios?

"Created by lucasdelatorre on 15/04/15"

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Utils.java:
```diff
@@ -0,0 +1,34 @@
+        return matcher.matches();
+    }
+
+    public static void showToast(CharSequence text, Context context)
```
En los métodos lo mismo. En este caso quedaria:

public static void showToast(CharSequence text, Context context) {
}

jpdelorenzo replied:
Por otro lado, te conviene que este método reciba el id del string y directamente lo obtenes aca. Asi no tenes que buscarlo en todos lados en el que lo llames.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Utils.java:
```diff
@@ -0,0 +1,34 @@
+    public static void showToast(CharSequence text, Context context)
+    {
+        int duration = Toast.LENGTH_SHORT;
+
```
Toast.makeText(context, text, Toast.LENGTH_SHORT).show();

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,117 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        TextView tv = (TextView) findViewById(R.id.tv_terms);
```
Para organizarnos solemos tener los siguientes métodos. setUi, init, populate y setListeners.  

En el primero hacemos los find en la view y se lo asginamos a las variables. En el init, solemos inicializar las cosas que necesitemos. En populate hacemos los setText, etc. Y por ultimo seteamos los listeners.

Por otro lado, defini las variables de la activity o fragment como privadas y que sean de instancia.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,117 @@
+        setContentView(R.layout.activity_login);
+
+        TextView tv = (TextView) findViewById(R.id.tv_terms);
+        tv.setText(Html.fromHtml(getString(R.string.login_terms)));
```
Este texto ponelo directo en el xml.

Lucheti29 replied:
No te estaría entendiendo este comentario.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,117 @@
+
+        Resources res = this.getResources();
+
+        if(username.isEmpty())
```
acordate lo del if

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,117 @@
+
+        if(username.isEmpty())
+        {
+            Utils.showToast(res.getString(R.string.login_emptyuser), getApplicationContext());
```
En vez de pasar el application context podrias directamente pasarle this

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,117 @@
+            return;
+        }
+
+        MessageType type = checkCredentials(username, pass);
```
Yo haria que el método checkCredencial solo devuelva true o false y los toast tambien los muestre este método. Asi te ahorras todos los ifs de abajo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,87 @@
+/**
+ * Created by lucasdelatorre on 15/04/15.
+ */
+public class Signup extends ActionBarActivity {
```
Lo mismo que te dije para la otra activity(lo de los cuatro métodos, variables, etc)

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/login_terms"
+        android:id="@+id/tv_terms"
```
Ponemos siempre primero el id, despues los atributos del layout.

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+            <ImageView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:id="@+id/iv_login_logo"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+                android:layout_height="wrap_content"
+                android:inputType="textEmailAddress"
+                android:ems="10"
+                android:id="@+id/et_login_user"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:ems="10"
+                android:id="@+id/et_login_pass"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+                android:layout_height="wrap_content"
+                android:text="@string/login_login"
+                android:layout_marginTop="20dp"
+                android:id="@+id/bn_login_login" />
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,98 @@
+                android:text="@string/login_signup"
+                android:background="@color/blue"
+                android:layout_marginTop="10dp"
+                android:id="@+id/bn_login_signup"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_signup.xml:
```diff
@@ -0,0 +1,62 @@
+    <EditText
+        android:layout_height="wrap_content"
+        android:layout_width="match_parent"
+        android:id="@+id/et_signup_user"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_signup.xml:
```diff
@@ -0,0 +1,62 @@
+    <EditText
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:id="@+id/et_signup_pass"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_signup.xml:
```diff
@@ -0,0 +1,62 @@
+    <EditText
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:id="@+id/et_signup_cpass"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_signup.xml:
```diff
@@ -0,0 +1,62 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/login_terms"
+        android:id="@+id/tv_signup_terms"
```
idem

jpdelorenzo commented on app/src/main/res/layout/activity_signup.xml:
```diff
@@ -0,0 +1,62 @@
+        android:layout_height="wrap_content"
+        android:text="@string/login_signup"
+        android:background="@color/bluewolox"
+        android:id="@+id/bn_signup_accept"
```
idem

Lucheti29 committed:
-Starter Activity removed: if the user is logged or not now it's handle by Login
-showToast method modified: better implementation
-Methods setUi, init, populate and setListeners added in Activities.
-'Ifs' with Wolox convention
-'id' attribute moved to top on its declaration
-Cleanup

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/Utils.java:
```diff
@@ -0,0 +1,28 @@
+
+    public static boolean validate(final String hex, final String stringPattern) {
+        Pattern pattern = Pattern.compile(stringPattern);
+        Matcher matcher;
```
Matcher matcher = pattern.matcher(hex);

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,136 @@
+
+public class Login extends Activity {
+
+    private TextView tvTerms;
```
Las variables privadas siempre las empezamos con m y las static con s. En este caso sería private TextView mTvTerms;

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,136 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if(isLogged()) {
```
if (isLogged()) startBoard();

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    private void attemptToLogin() {
+        String username = ((EditText) findViewById(R.id.et_login_user)).getText().toString();
```
estas variables tambien definilas a nivel clase y privadas :dancer:

Lucheti29 replied:
A nivel clase te referís como variables de clase o de instancia?

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,136 @@
+        String username = ((EditText) findViewById(R.id.et_login_user)).getText().toString();
+        String pass = ((EditText) findViewById(R.id.et_login_pass)).getText().toString();
+
+        if(username.isEmpty()) {
```
Espacio despues del if :ghost:

Lucheti29 replied:
No entiendo.

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,136 @@
+            return;
+        }
+
+        MessageType type = checkCredentials(username, pass);
```
creo que te lo había dicho pero me parece mejor que el checkCredentials solo devuelva un boolean y handlear los tipos de errores dentro del mismo método.

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,95 @@
+
+public class Signup extends ActionBarActivity {
+
+    private TextView tvTerms;
```
Lo mismo con las variables :dancers:

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,95 @@
+        //TODO: Register new user
+    }
+
+    private void getInputs()
```
Sin enter las {

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,95 @@
+
+    private void getInputs()
+    {
+        etUsername = ((EditText) findViewById(R.id.et_signup_user)).getText().toString();
```
los find en la view hacelos en el setUi :see_no_evil:

Lucheti29 replied:
A menos que esté entendiendo mal, no tiene sentido que los haga en setUi() porque getInputs() actualiza las variables con el contenido de los EditText (que pueden cambiar más de una vez). Si aún así llamara a setUi() por cada vez que clickee un botón, re-setearía cosas que no son necesarias.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,95 @@
+
+        if (etUsername.isEmpty()) {
+            Utils.showToast(this, R.string.signup_emptyuser);
+        }
```
} else if (....) {

Lucheti29 replied:
Listo.

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Signup.java:
```diff
@@ -0,0 +1,95 @@
+    private boolean userExists(String username) {
+        boolean value = false;
+
+        if (etUsername.isEmpty()) {
```
Hacete un método para todas las validaciones

Lucheti29 replied:
No te estaría entendiendo.

jpdelorenzo commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -0,0 +1,92 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/login_terms"
+        android:layout_alignParentBottom="true"
```
Acordate despues del id, todos los attributes del layout

Lucheti29 replied:
Listo.

Lucheti29 committed:
-Pull request corrections

jpdelorenzo commented on app/src/main/java/ar/com/wolox/lucasdelatorre/training/activities/Login.java:
```diff
@@ -0,0 +1,122 @@
+        //When the backend is implemented, it returns true
+    }
+
+    private boolean validateInputs()
```
Acordate lo de las { :dancer:

Lucheti29 replied:
Es más fuerte que yo ese 'Enter'. Corrigiendo esto, ¿ya puedo mergear a develop? Sino se me va a complicar para avanzar con el backend.

Lucheti29 replied:
Listo.

Lucheti29 committed:
-Pull request correction

jpdelorenzo commented:
:+1:
